keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -85,6 +86,17 @@ class ResourceActuator(
         return@withTracingContext
       }
 
+      val deliveryConfig = deliveryConfigRepository.deliveryConfigFor(resource.id)
+      val environment = checkNotNull(deliveryConfig.environmentFor(resource)) {
+        "Failed to find environment for ${resource.id} in deliveryConfig ${deliveryConfig.name}"
+      }
+
+      if (deliveryConfig.isPromotionCheckStale()) {
+        log.debug("Environment check for {} is {} old, skipping checks", id, deliveryConfig.isPromotionCheckStale())
+        publisher.publishEvent(ResourceCheckSkipped(resource.kind, id, "PromotionCheckStale"))
+        return@withTracingContext
+      }
```

Comment:
Skip resource check if environment has not been checked recently. This is to allow for environment checks to catch up if the application has been paused, and we'll therefore deploy an up-to-date artifact rather than whatever was approved last time the app was un-paused. If manual intervention has happened during the pause time we could roll back to an unwanted older version.

Reply:
Done